### PR TITLE
vcat with all nulls preserves isordered

### PR DESCRIPTION
**Pull request by gustafsson**
Based on #28

### CODE REVIEW

**Comment by nalimilan** [comment]
I would also check that levels are empty. Even if in practice it shouldn't make a big difference, that sounds more correct.

File: src/array.jl [hunk 1]
```diff
@@ -364,15 +364,16 @@ arraytype(A::CatArray...) = NullableCategoricalArray
 
 function vcat(A::CatArray...)
     newlevels, ordered = mergelevels(map(levels, A)...)
+    ordered &= any(isordered, A)
```
Comment (nalimilan) [inline src/array.jl]: This line isn't needed now, right?

File: test/12_nullablearray.jl [hunk 2]
```diff
@@ -999,13 +1007,17 @@ r = vcat(ca1, ca2)
...
-@test !isordered(r)
+@test isordered(r)
 
-ordered!(ca2,true)
+ca1 = NullableCategoricalArray(["a", "b"], [false, true])
+ca2 = NullableCategoricalArray{String}(2)
+ordered!(ca1, true)
+@test isempty(levels(ca2))
 r = vcat(ca1, ca2)
+@test isequal(r, NullableCategoricalArray(["a", "", "", ""],[false, true, true, true]))
```
Comment (nalimilan) [inline test/12_nullablearray.jl]: Inconsistent spacing here.

File: src/array.jl (hunk 1 above)
Comment (gustafsson) [inline src/array.jl]: This PR does so that adding an array without any levels wouldn't conflict with any existing ordering. But vcat of arrays with only nulls (i.e no levels anywhere) would then be considered ordered even if they all had isordered==false, that was not the intent. This row covers that edge case.

**Comment by gustafsson** [comment]
I added the tests you asked for

File: src/array.jl (hunk 1 above)
Comment (nalimilan) [inline src/array.jl]: OK. But then if `ordered` is false, you're doing the check unnecessarily (same for the one below). Better write `ordered = ordered && any(...) && all(...)`.

**Comment by nalimilan** [comment]
Could you just change the `ordered` condition? Would be cool to merge this and tag a new release soon.

**Comment by nalimilan** [comment]
OK, I'll change that after the fact. Thanks again!

**Comment by gustafsson** [comment]
You're welcome :)
